src/server/game/DataStores/DB2Stores.cpp

```diff
@@ -1510,8 +1525,27 @@ std::set<uint32> DB2Manager::GetItemBonusTree(uint32 itemId, uint32 itemBonusTre
             continue;
 
         for (ItemBonusTreeNodeEntry const* bonusTreeNode : treeItr->second)
-            if (bonusTreeNode->BonusTreeModID == itemBonusTreeMod)
+        {
+            if (bonusTreeNode->BonusTreeModID != itemBonusTreeMod)
+                continue;
+            
+            if (bonusTreeNode->BonusListID)
+            {
                 bonusListIDs.insert(bonusTreeNode->BonusListID);
+            }
+            else if (bonusTreeNode->ItemLevelSelectorID)
+            {
+                ItemLevelSelectorEntry const* selector = sItemLevelSelectorStore.LookupEntry(bonusTreeNode->ItemLevelSelectorID);
+                if (!selector)
+                    continue;
+
+                int16 levelChange = int16(selector->ItemLevel) - proto->ItemLevel;
+                
+                auto itr = _selectorToBonus.find(levelChange);
```

Comment:
This is wrong. You are supposed to use ItemBonusListLevelDelta here (no need for custom filled container based on hardcoded ids)

Reply:
Ah damn, no idea how I missed that file. I wasn't happy with _selectorToBonus at all, so thanks for the info!